v1/feed.js

```diff
@@ -0,0 +1,167 @@
+        });
+    });
+
+    return P.props(props);
+
+};
+
+
+Feed.prototype.aggregated = function(hyper, req) {
+
+    var self = this;
+    var rp = req.params;
+    var date;
+    var dateArr;
+
+    // key the records on the date in the format YYYY-MM-DD
+    try {
+        date = new Date(Date.UTC(rp.yyyy, rp.mm - 1, rp.dd));
+        date = date.toISOString().split('T').shift();
+        dateArr = date.split('-');
+    } catch (err) {
+        return P.reject(new HTTPError({
+            status: 400,
+            body: {
+                type: 'bad_request',
+                description: 'wrong date format specified'
+            }
+        }));
+    }
+
+    // check if we have a record in Cassandra already
+    return hyper.get({
+        uri: new URI([rp.domain, 'sys', 'key_value', 'feed.aggregated', date])
+    }).then(function(res) {
+        // we've got a cache hit, so we just need to request
+        // the random component and return the bundle
+        if (!res.body) {
+            return P.reject(new HTTPError({
+                status: 500,
+                body: {
+                    type: '#internal_error',
+                    detail: 'No data received for aggregated feed date ' + date,
+                    feed_date: date
+                }
+            }));
+        }
+        return self._makeFeedRequests(['random'], hyper, rp).then(function(rndRes) {
+            res.body.random = rndRes.random.body;
+            // make a new ETag since we are changing a part of the body
+            res.headers.etag = dateArr.join('') + '/' + uuid.now().toString();
+            return res;
+        }).catch(function(e) {
+            // something went wrong while retrieving the random part of
+            // the response from MCS, so just return the stored content
+            // as there is no need to error out for this edge case
+            return res;
+        });
+    }).catch({ status: 404 }, function(err) {
+        // it's a cache miss, so we need to request all
+        // of the components and store them
+        return self._makeFeedRequests(Object.keys(FEED_URIS), hyper, rp, dateArr)
+        .then(function(result) {
+            // assemble the final response to be returned
+            var finalResult = {
+                status: 200,
+                headers: {
+                    'cache-control': self.options.feed_cache_control,
+                    // mimic MCS' ETag value
+                    etag: dateArr.join('') + '/' + uuid.now().toString(),
+                    // TODO: need a way to dynamically derive this
+                    'content-type': 'application/json; charset=utf-8; ' +
+                        'profile="https://www.mediawiki.org/wiki/Specs/aggregated-feed/0.5.0"'
+                },
+                body: {}
+            };
+            // populate its body
+            Object.keys(result).forEach(function(key) {
+                finalResult.body[key] = result[key].body;
+            });
```

Comment:
`Object.assign(finalResult.body, result.body)`.. Or, actually, why do we need to copy it?